lightkurve/mast.py

```diff
@@ -61,86 +69,242 @@ def search_kepler_products(target):
+                            '(radius currently set to {} arcseconds)'
+                            ''.format(targetlimit, len(np.unique(ids)), radius))
+        okids = ids[np.sort(np.unique(ids, return_index=True)[1])[0:targetlimit]]
+        mask = np.zeros(len(ids), dtype=bool)
+        idsort = np.zeros(len(ids))
+        for idx, o in enumerate(okids):
+            p = ids == o
+            idsort[p] = int(idx)
+            mask |= p
+        products['ids'] = idsort
+        products = products[mask]
+
+    #For Kepler short cadence data there are additional rules, so find anywhere
+    #where there is short cadence data...
+    scmask = np.asarray(['Short' in d for d in products['description']]) &\
+             np.asarray(['kplr' in d for d in products['dataURI']])
+    if np.any(scmask):
+        #Error check the user if there's multiple months and they didn't ask
+        #for a specific one
+        if month is None:
+            raise ArchiveError("Found {} different Target Pixel Files "
+                               "for target {} in Quarter {}. "
+                               "Please specify the month (1, 2, or 3)."
+                               "".format(len(products), target, quarter))
+        #Get the short cadence date lookup table.
+        table = ascii.read('/Users/ch/K2/repos/lightkurve/lightkurve/data/short_cadence_month_lookup.csv')
+        #Grab the dates of each of the short cadence files and make sure they
+        #all use the correct month
+        finalmask = np.ones(len(products), dtype=bool)
+        for c in np.unique(products[scmask]['qoc']):
+            ok = (products['qoc'] == c) & (scmask)
+            mask = np.zeros(np.shape(products[ok])[0], dtype=bool)
+            for m in month:
+                udate = (table['Date'][np.where((table['Month'] == m) & (table['Quarter'] == c))[0][0]])
+                mask |= np.asarray(products['dates'][ok]) == udate
+            finalmask[ok] = mask
+        products = products[finalmask]
+        #Sort by id, then date and quarter
+        products.sort(['ids', 'dates', 'qoc'])
+        if len(products) < 1:
+            raise ArchiveError("No {} File found for {} "
+                               "at month {} at MAST.".format(filetype, target, month))
+
+    #If there is no specified quarter but there are many campaigns/quarters
+    #returned, throw an error to the user
+    if (len(np.unique(products['qoc'])) > 1) & (campaign is None) & (quarter is None):
+        raise ArchiveError("Found {} different Target Pixel Files "
+                           "for target {}. Please specify quarter/month "
+                           "or campaign number."
+                           "".format(len(products), target))
+
+    #Otherwise download all the files
+    if verbose:
+        print('Found {} File(s)'.format(np.shape(products)[0]))
+    if not verbose:
+        old_stdout = sys.stdout
+        sys.stdout = open(os.devnull, "w")
```

Comment:
Is there a reason to re-assign `sys.stdout` *and* do `astropylog.setLevel`?

Reply:
I guess not

Reply:
The sys.stdout doesn't remove astropy.log messages